### ISSUE TITLE
Request For Trials

### ISSUE THREAD

**Issue by lotfyman95** [Anon-Exploiter/ine-dl#14]
got an Error message while it was fetching paid subscription courses while I was on a trial

```bash
IndexError: list Index out of Range

'MainThread' (20720):
Traceback (most recent call last):

> File "D:\ine-dl\ine.py", line 1074, in <module>
    main()
    └ <function main at 0x000002C49DCD18A0>

  File "D:\ine-dl\ine.py", line 1026, in main
    user_subscriptions = fetch_user_subscriptions(all_course_contents, courses_with_access, all_courses_metadata)
                         │                        │                    │                    └ 'all_courses_metadata.json'
                         │                        │                    └ 'all_courses_with_access.json'
                         │                        └ [{'id': 'bfba41be-429a-11e4-a79f-22000b3582a3', 'tags': ['CCNA', 'Cisco', 'Security'], 'images': {'main_image_url': 'https://...
                         └ <function fetch_user_subscriptions at 0x000002C49DCD1120>

  File "D:\ine-dl\ine.py", line 306, in fetch_user_subscriptions
    subscriptions = json.loads(user_subs.text).get("data")[0].get("passes").get("data")
                    │    │     │         └ <property object at 0x000002C49DA0D490>
                    │    │     └ <Response [200]>
                    │    └ <function loads at 0x000002C49D366AC0>
                    └ <module 'json' from 'C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.12_3.12.496.0_x64__qbz5n2kfra8p0\\Lib\...

IndexError: list index out of range
```

**Comment by Anon-Exploiter** [comment]
If it's possible, pass me the account. I'll take a look. 

Can you access courses just fine online?

**Comment by lotfyman95** [comment]
I can access courses online Normally, 
Unfortunately this is not possible Since it's Business Related Credintials

**Comment by lotfyman95** [comment]
it appears after using the latest version of the @Jayapraveen code that I do not have the subscription/pass to Download the courses although I can browse freely and play the videos, However, in the @Jayapraveen script I can list the all courses. believe if the listing was tied to subscription authorization it wouldn't have worked.

**Comment by LOLCATATONIA** [comment]
I am also receiving this error on Ubuntu 22.04.1 LTS.

Business credentials as well. Can I help debug somehow?

Yes, seems like line 297 doesn't work for us. https://subscriptions.ine.com/subscriptions/subscriptions returns:
{"data":[],"meta":{"cursor":{"current":1,"prev":null,"next":null,"count":0}}}

**Comment by Anon-Exploiter** [comment]
> I am also receiving this error on Ubuntu 22.04.1 LTS.
> 
> Business credentials as well. Can I help debug somehow?
> 
> Yes, seems like line 297 doesn't work for us. https://subscriptions.ine.com/subscriptions/subscriptions returns: {"data":[],"meta":{"cursor":{"current":1,"prev":null,"next":null,"count":0}}}

Business APIs might differ a bit, happy to share creds?

**Comment by Anon-Exploiter** [comment]
@lotfyman95 this is now fixed. For any further comments or issues regarding the fix, use the issue #17. 

https://github.com/Anon-Exploiter/ine-dl/issues/17#issuecomment-2093992591
